### PR TITLE
Swap functionality

### PR DESCRIPTION
**Pull request by CedThen**

### CODE REVIEW

File: src/components/SwipeableItem.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const {
 
 interface SwipeRowProps {
     swipeThreshold: number;
-    onSwipe: (taskId: number) => void;
+    onSwipe: (TaskId: number) => void;
```
Comment (showduhtung) [inline src/components/SwipeableItem.tsx]: arguments are usually not capitalized. is there a reason for this?

File: src/components/TaskList/TaskListItem.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ interface TaskListItemProps {
 
 export const TaskListItem = (props: TaskListItemProps) => {
```
Comment (showduhtung) [inline src/components/TaskList/TaskListItem.tsx]: can this be de-structured?

File: src/components/TaskList/TaskListItem.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ interface TaskListItemProps {
...
     const dispatch = useDispatch();
-    const date: Date = new Date(props.item.date.replace(/-/g, '/'));
+    const date: Date = new Date(props.item.date);
```
Comment (showduhtung) [inline src/components/TaskList/TaskListItem.tsx]: it'll be `= new Date(item.date)`

File: src/screens/Drawer/NewAssignmentsScreen.tsx [hunk 4]
```diff
@@ -43,8 +43,8 @@ export const NewAssignmentsScreen = (props: NewAssignmentsScreenProps) => {
         </TouchableOpacity>
     );
 
-    const deleteItem = (taskId) => {
-        const updatedData = data.filter((d) => d.taskId !== taskId);
+    const deleteItem = (TaskId) => {
+        const updatedData = data.filter((d) => d.TaskId !== TaskId);
```
Comment (showduhtung) [inline src/screens/Drawer/NewAssignmentsScreen.tsx]: what you had previously is more conventional

File: src/screens/Drawer/TaskDetailsScreen.tsx [hunk 5]
```diff
@@ -21,6 +24,7 @@ export const TaskDetailsScreen = (props) => {
 
     const closeModalHandler = () => {
         setModalVisible(false);
+        dispatch(resetSwapConfig());
```
Comment (showduhtung) [inline src/screens/Drawer/TaskDetailsScreen.tsx:29]: no need for loading logic?

File: src/screens/RequestSwap/SwapConfirmation.tsx [hunk 6]
```diff
@@ -1,7 +1,19 @@
...
 import { View, StyleSheet } from 'react-native';
-import { Layout, Text } from '@ui-kitten/components';
+import { Layout, Text, Button } from '@ui-kitten/components';
+import { useSelector } from 'react-redux';
+import { loadStateActionTypes } from '../../store/actions';
 
-export const SwapConfirmation = () => {
-    return <Text>Request Sent!</Text>;
+export const SwapConfirmation = (props) => {
+    const requestState = useSelector((state) => state.loadStateReducer.loadStatus);
```
Comment (showduhtung) [inline src/screens/RequestSwap/SwapConfirmation.tsx:8]: is this in a modal? if so, then this state should be passed through props

Comment (showduhtung) [inline src/screens/RequestSwap/SwapConfirmation.tsx]: is this in a modal? if so, then this state should be passed through props

File: src/screens/RequestSwap/SwapScreen.tsx [hunk 7]
```diff
@@ -22,7 +24,26 @@ export const SwapScreen = (props: SwapScreenProps) => {
     //props.route.params.selectedOption stores what was selected on previous screen
     const selectedDate = useSelector((state) => state.calendarReducer.selectedDate.date);
     const [checked, setChecked] = useState<boolean>(false);
-    const [selectedIndex, setSelectedIndex] = useState<IndexPath>(0);
+    const [selectedIndex, setSelectedIndex] = useState(null);
+    // const [nextClickable, setNextClickable] = useState<boolean>(false);
+    //this is the date they select to swap on
+    const selectedSwapDate = useSelector((state) => state.swapReducer.swapDate);
+    const dispatch = useDispatch();
+
```
Comment (showduhtung) [inline src/screens/RequestSwap/SwapScreen.tsx:32]: should the swap state be here?

File: src/screens/RequestSwap/SwapSummary.tsx [hunk 8]
```diff
@@ -1,21 +1,67 @@
...
+        (state) => state.swapReducer,
+    );
+
     const onConfirmPress = () => {
-        props.navigation.navigate('SwapConfirmation');
+        dispatch(sendSwapRequest(swapOption, swapDate, swapTarget));
+        // props.navigation.navigate('SwapConfirmation');
     };
 
+    const swapTargetString =
```
Comment (showduhtung) [inline src/screens/RequestSwap/SwapSummary.tsx]: it's a little redundant to use swap as a prefix to constants particularly in this container file. `target` and `targetDate` or something along the lines could suffice

File: src/store/actions/swapActions.ts [hunk 9]
```diff
@@ -1,22 +1,73 @@
...
+        },
+        {
+            headers: {
+                authorization: accesskey,
+            },
+        },
+    );
+}
+
+export const sendSwapRequest = (swapOption, swapDate, swapTarget) => {
```
Comment (showduhtung) [inline src/store/actions/swapActions.ts]: using swap in argument is pretty redundant. `(option, date, target)` is better

File: src/store/reducers/swapReducer.ts [hunk 10]
```diff
@@ -1,16 +1,41 @@
...
+import {
+    SELECT_SWAP_DATE,
+    SELECT_SWAP_OPTION,
+    SELECT_SWAP_TARGET,
+    SEND_SWAP_REQUEST,
+    RESET_SWAP_CONFIG,
+} from '../actions';
 
 const initialState = {
+    swapOption: 0,
```
Comment (showduhtung) [inline src/store/reducers/swapReducer.ts]: seeing that the reducer is called `swapReducer`, `swapReducer.date || swapReducer.target || swapReducer.option` is already descriptive enough

**Review by showduhtung — CHANGES_REQUESTED** [review]

File: src/components/TaskList/TaskListItem.tsx (hunk 2 above)
Comment (CedThen) [reply src/components/TaskList/TaskListItem.tsx]: ok

File: src/screens/Drawer/TaskDetailsScreen.tsx (hunk 5 above)
Comment (CedThen) [reply src/screens/Drawer/TaskDetailsScreen.tsx:29]: no, this is just to clear reducer of any data from user actions on modal close

File: src/store/actions/swapActions.ts (hunk 9 above)
Comment (CedThen) [reply src/store/actions/swapActions.ts]: ok

File: src/store/reducers/swapReducer.ts (hunk 10 above)
Comment (CedThen) [reply src/store/reducers/swapReducer.ts]: changes made

File: src/screens/RequestSwap/SwapSummary.tsx (hunk 8 above)
Comment (CedThen) [reply src/screens/RequestSwap/SwapSummary.tsx]: okie

**Pull request merged**